src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestStream.cs

```diff
@@ -100,7 +100,7 @@ public override void Write(byte[] buffer, int offset, int count)
                 throw new ArgumentOutOfRangeException("count");
             }
 
-            if (offset + count > buffer.Length)
+            if ((long)offset + (long)count > (long)buffer.Length)
```

Comment:
I believe such overflows can be handled without the casts just by changing the shape of the expression:
```C#
if (count > buffer.Length - offset)
```